src/leveled_sst.erl

```diff
@@ -1395,41 +1394,30 @@ fetch(LedgerKey, Hash,
 %% Fetch pointers to the slots the SST file covered by a given key range.
 fetch_range(StartKey, EndKey, Summary, FilterFun, true) ->
     {Slots, RTrim} =
-        lookup_slots(
-            StartKey,
-            EndKey,
-            Summary#summary.index,
-            FilterFun),
+        lookup_slots(StartKey, EndKey, Summary#summary.index, FilterFun),
```

Comment:
Not sure worth it, but you can simplify the code a bit by 
```
   Lastkey = if not RTrim -> all; RTrim -> EndKey end,
 ```
 And use LastKey below without the need to do a case on `RTrim`.

Reply:
Updated